core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -66,8 +78,45 @@
     public final static String INDEX = ".ingest";
     public final static String TYPE = "pipeline";
 
+    final static Settings INGEST_INDEX_SETTING = Settings.builder()
+        .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+        .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+        .put("index.mapper.dynamic", false)
+        .build();
+
+    final static String PIPELINE_MAPPING;
+
+    static {
+        try {
+            PIPELINE_MAPPING = XContentFactory.jsonBuilder().startObject()
+                .field("dynamic", "strict")
+                .startObject("_all")
+                    .field("enabled", false)
+                .endObject()
+                .startObject("properties")
+                    .startObject("processors")
```

Comment:
I opened #15842 to discuss this further. I think we can move the cluster state now, before was tricker with they way how things were designed.